qiskit/compiler/transpiler.py

```diff
@@ -669,6 +677,12 @@ def _parse_transpile_args(
     }
 
     list_transpile_args = []
+    if not ignore_backend_supplied_default_methods:
+        if scheduling_method is None and hasattr(backend, "get_scheduling_stage_method"):
+            scheduling_method = backend.get_scheduling_stage_method()
+        if translation_method is None and hasattr(backend, "get_translation_stage_method"):
+            translation_method = backend.get_translation_stage_method()
+
```

Comment:
Well the list of potential names here will never exceed the default stages for a `StagedPassManager` which is what `transpile()` and the plugin interface is limited to (see https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/transpiler/preset_passmanagers/plugin.py#L34 and https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/transpiler/passmanager.py#L397-L402). But, yeah I started small here to start on purpose to limit the exposed API surface to just what seems likely for a backend to need these hook points for. But if we did expand it we'd be limited to the built-in/default stages as this is just for `transpile()` which has more predefined structure. The list of default stages can expand over time (see the note in: https://qiskit.org/documentation/stubs/qiskit.transpiler.StagedPassManager.html ) but it doesn't mean we'll allow arbitrary fields for these names. If people require the interface with lower level control like that they can create custom `StagedPassManagers` and if a backend wants to do that too they can leverage that interface which enables it. For the hook points in this PR it's more about enabling backends to influence `transpile()` at fixed points in the pipeline to better support custom hardware-specific requirements we can't model in the current interface.

Reply:
I think for right now it's ok since it is always going to be a limited set of stages supported. I think when/if it's > 2 doing this in something a bit more compact with a loop makes a ton of sense. I'd put a TODO comment here about that in the future (or just do it now, but that would increase the amount of lines touched to do the refactor for the other methods). But I think we'll be able to remove most of this code in the medium term. As part of 0.23.0 I want to deprecate the broadcast support in the arguments on `transpile()` towards being able to close: https://github.com/Qiskit/qiskit-terra/issues/7741 once that's done we'll want to refactor a lot of this argument handling anyway as it's only needed here to support broadcasting. Once that's done `transpile()` will basically become:

```python
from qiskit.transpiler.preset_passmanagers import generate_preset_pass_manager

def transpile(circuits, backend, optimization_level=1, **transpile_kwargs):
    pm = generate_preset_pass_manager(optimization_level, backend, **transpile_kwargs)
    return pm.run(circuits)
```

and we can do this unified handling directly in the `pass_manager_config` more directly.